lib/api/contracts/model_contract.rb

```diff
@@ -42,12 +42,27 @@ def self.attribute_validations
       end
 
       def self.attribute(*attributes, &block)
+        attributes.each do |attribute|
+          property attribute
+        end
+
         writable_attributes.concat attributes.map(&:to_s)
         if block
           attribute_validations << block
         end
       end
 
+      # we want to add a validation error whenever someone sets a property that we don't know.
+      # However AR will cleverly try to resolve the value for errorneous properties. Thus we need
+      # to hook into this method and return nil for unknown properties to avoid NoMethod errors...
+      def read_attribute_for_validation(attribute)
+        if respond_to? attribute
+          send attribute
+        end
+
+        nil
```

Comment:
Thanks for spotting that.
I am actually not sure whether this method will even be called in a place where I "care", but I thought I probably should. And I did so by providing a flawed implementation ^^